### PR TITLE
add setting: export.pdfPageSize

### PR DESCRIPTION
**Pull request by tessus**
adds a setting to specify the page size for exporting notes to PDF

### CODE REVIEW

**Comment by tessus** [comment]
>  Shouldn't these settings also be applied when printing to paper?

I thought about it, but the printer subsystem already takes the settings from the OS, so there's no need to do that. In fact this could actually create a problem. Imagine the following situation: Someone uses the portable version of Joplin and prints on system A. System A uses a printer with A4. Now the user switches to system B and the attached printer uses Letter or whatever. But Joplin would print A4 on paper format X on system B, if we had applied the setting to printing as well.

> Also since there's two properties, why not create two setting values, pageSize and pageOrientation?

Sure, I can do that. The problem is that it's impossible to allgn the text (keys and values), unless I have overlooked something. But yes, from a technical perspective, 2 settings are more straight forward and definitely more logical.
I'm more than happy to create 2 settings.

**Comment by tessus** [comment]
Btw, I can also add other paper formats. These are the ones that are supported by Electron's printToPDF: A3, A4, A5, Legal, Letter, Tabloid

File: ReactNativeClient/lib/models/Setting.js [hunk 1]
```diff
@@ -407,6 +407,19 @@ class Setting extends BaseModel {
...
+					'A4': _('A4'),
+					'Letter': _('Letter'),
+				};
+			}},
+			'export.pdfPageOrientation': { value: false, type: Setting.TYPE_BOOL, isEnum: true, public: true, appTypes: ['desktop'], label: () => _('Page orientation for PDF export'), options: () => {
+				return {
+					false: _('Portrait'),
+					true: _('Landscape'),
+				};
+			}},
```
Comment (laurent22) [inline ReactNativeClient/lib/models/Setting.js]: I know there's only two values but I find it a bit confusing that it's true/false. Could you change it to a string enum with values "portrait" and "landscape"?

Then when calling printToPdf you can just do `landscape: Setting.value('export.pdfPageOrientation') === 'landscape'`

Comment (tessus) [reply ReactNativeClient/lib/models/Setting.js]: Yes, will do, if you really think this is better. I'll be changing it in the next 30 minutes.

Comment (tessus) [reply ReactNativeClient/lib/models/Setting.js]: Done.

**Comment by laurent22** [comment]
> Btw, I can also add other paper formats. These are the ones that are supported by Electron's printToPDF: A3, A4, A5, Legal, Letter, Tabloid

Sorry I had missed that. Indeed could you add them all please? (may as well since we get that for free with Electron)

**Comment by tessus** [comment]
> Sorry I had missed that. Indeed could you add them all please? (may as well since we get that for free with Electron)

Sure, I'm just not so sure about the order. But I will order them like this:

A4, Letter, A3, A5, Tabloid, Legal

A4 and Letter are the most common ones. Then I go from big to small (first DIN, then US).

**Comment by tessus** [comment]
Done.

**Comment by tessus** [comment]
Hmm, the CI build failure makes no sense.

**Comment by laurent22** [comment]
> Hmm, the CI build failure makes no sense.

That's me, I forgot to update a package on the cli app.

**Comment by laurent22** [comment]
Thanks for the PR @tessus!

**Pull request merged**